src/lib.rs

```diff
@@ -86,11 +86,18 @@ mod up_core_api {
 // Types not used by up_rust, but re-exported to up_rust users, keeping them in their respective submodules
 #[cfg(feature = "udiscovery")]
 pub use up_core_api::udiscovery;
-#[cfg(feature = "usubscription")]
-pub use up_core_api::usubscription;
 #[cfg(feature = "utwin")]
 pub use up_core_api::utwin;
 
+mod core;
+#[cfg(feature = "usubscription")]
+pub use core::{
```

Comment:
I would rather not re-export all of these type but keep them in module `core::usubscription`. Anticipating that we will be doing a similar thing next for `utwin` and `udsicovery`, I think we should keep at least some structure.